ansible/roles/cni/tasks/flannel.yml

```diff
@@ -1,4 +1,5 @@
 ---
-- name: Applying Flannel (1/1)
-  become: true
-  command: kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml
+- name: applying flannel
```

Comment:
Flannel by memory needed some additional fields passed into the kubeadm init function. This was something not quite finished when we added flannel previously, perhaps we can raise this as tech debt?

```
NOTE: If kubeadm is used, then pass --pod-network-cidr=10.244.0.0/16 to kubeadm init to ensure that the podCIDR is set.
```
source: https://coreos.com/flannel/docs/latest/kubernetes.html

Reply:
Added a check which will fail to apply flannel when k8s is not setup using the pod network cidr option.

Besides that, its the user's responsibility to correctly set a pod network cidr when they choose flannel, as the default is now calico.